File: tests/fixtures/simple_python/handlers.py
"""HTTP-style handlers."""
from auth import authenticate, verify_token
from models import Order, User


def get_user(user_id: int, token: str) -> dict:
    user = authenticate(user_id, token)
    if user is None:
        return {"error": "unauthorized"}
    return {"id": user.user_id, "name": user.display_name()}


def create_order(user_id: int, token: str, items: list[str]) -> dict:
    if not verify_token(token):
        return {"error": "invalid token"}
    user = User(user_id=user_id, name="buyer")
    order = Order(order_id=1, user=user)
    for item in items:
        order.add_item(item)
    return {"order_id": order.order_id, "items": order.total_items()}

File: tests/fixtures/simple_typescript/handlers.ts
// HTTP-style handlers.
import { authenticate, verifyToken } from "./auth";
import { Order, User } from "./models";

export function getUser(userId: number, token: string): Record<string, unknown> {
  const user = authenticate(userId, token);
  if (user === null) {
    return { error: "unauthorized" };
  }
  return { id: userId, name: user.displayName() };
}

export function createOrder(
  userId: number,
  token: string,
  items: string[]
): Record<string, unknown> {
  if (!verifyToken(token)) {
    return { error: "invalid token" };
  }
  const user = new User(userId, "buyer");
  const order = new Order(1, user);
  for (const item of items) {
    order.addItem(item);
  }
  return { orderId: order.orderId, items: order.totalItems() };
}
